### PR TITLE
Bug 2059255: Fix stale metrics by resetting them

### PR DESCRIPTION
**Pull request by gnufied**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=2059255

### CODE REVIEW

File: pkg/check/node_hw_version_test.go [hunk 1]
```diff
@@ -51,15 +69,14 @@ vsphere_node_hw_version_total{hw_version="vmx-15"} 1
...
-			err = customizeVM(ctx, node, &types.VirtualMachineConfigSpec{
-				ExtraConfig: []types.BaseOptionValue{
-					&types.OptionValue{
-						Key: "SET.config.version", Value: test.hwVersion,
-					},
-				}})
-			if err != nil {
-				t.Fatalf("Failed to customize node: %s", err)
+			if len(test.hwVersions) > 0 {
+				for i := range test.hwVersions {
```
Comment (bertinatto) [inline pkg/check/node_hw_version_test.go:73]: nit: maybe you can just remove the if clause above

Comment (gnufied) [reply pkg/check/node_hw_version_test.go:73]: that sounds fine, but this is PR is a cherry-pick of master branch and hence I am bit hesitant to modify it here.

**Comment by jsafrane** [comment]
Lgtm to the code. But why did you cherry-pick merge commits? I can see it nicely shows from which PR it comes from, still, it's something I've never seen in the other cherry picks.

**Comment by gnufied** [comment]
Moved the commits to individual commits. PTAL.

**Pull request merged**